cmd/snapd-apparmor/main_test.go

```diff
@@ -0,0 +1,182 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	snapd_apparmor "github.com/snapcore/snapd/cmd/snapd-apparmor"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/testutil"
+)
+
+// Hook up check.v1 into the "go test" runner
+func Test(t *testing.T) { TestingT(t) }
+
+type mainSuite struct {
+	testutil.BaseTest
+}
+
+var _ = Suite(&mainSuite{})
+
+func (s *mainSuite) SetUpTest(c *C) {
+	dirs.SetRootDir(c.MkDir())
+
+}
+
+func (s *mainSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
+}
+
+func (s *mainSuite) TestIsContainerWithInternalPolicy(c *C) {
+	c.Assert(snapd_apparmor.IsContainerWithInternalPolicy(), Equals, false)
+
+	appArmorSecurityFSPath := filepath.Join(dirs.GlobalRootDir, "/sys/kernel/security/apparmor/")
+	err := os.MkdirAll(appArmorSecurityFSPath, 0755)
+	c.Assert(err, IsNil)
+
+	c.Assert(snapd_apparmor.IsContainerWithInternalPolicy(), Equals, false)
+
+	// simulate being inside WSL
+	testutil.MockCommand(c, "systemd-detect-virt", "echo wsl")
+	c.Assert(snapd_apparmor.IsContainerWithInternalPolicy(), Equals, true)
+
+	// simulate being inside a container environment
+	testutil.MockCommand(c, "systemd-detect-virt", "echo lxc")
+	c.Assert(snapd_apparmor.IsContainerWithInternalPolicy(), Equals, false)
+
+	f, err := os.Create(filepath.Join(appArmorSecurityFSPath, ".ns_stacked"))
+	c.Assert(err, IsNil)
+	f.WriteString("yes")
+	f.Close()
+	c.Assert(snapd_apparmor.IsContainerWithInternalPolicy(), Equals, false)
+
+	f, err = os.Create(filepath.Join(appArmorSecurityFSPath, ".ns_name"))
+	c.Assert(err, IsNil)
+	defer f.Close()
```

Comment:
(nitpick) This Close() is inconsistent with the Close() above, but ioutil.WriteFile would make it go